helix-lsp/src/client.rs

```diff
@@ -268,6 +274,93 @@ impl Client {
             .expect("language server not yet initialized!")
     }
 
+    #[inline] // TODO inline?
+    pub fn supports_feature(&self, feature: LanguageServerFeature) -> bool {
+        let capabilities = match self.capabilities.get() {
+            Some(capabilities) => capabilities,
+            None => return false, // not initialized, TODO unwrap/expect instead?
+        };
```

Comment:
```suggestion
        let Some(capabilities) = match self.capabilities.get() else {
            return false; // not initialized, TODO unwrap/expect instead?
        };
```

Reply:
rustfmt won't help us fix this yet but I think the style guide suggests (https://github.com/rust-lang/rust/blob/c8ead2e693a22fe94c6b3edeb3f49c7e6aec3912/src/doc/style-guide/src/statements.md#single-line-let-else-statements):

```rs
let Some(capabilities) = self.capabilities.get() else { return false };
```

Reply:
Yeah `let .. else` is better, though as the `TODO` notes, I'm not sure if an `expect` or just `Client::capabilities()` may be better here, because this method shouldn't be run if the `Client` isn't initialized yet, and I think this can produce weird issues if features are supported (although the client does) but the client isn't initialized, and it's easier to find bugs if helix just terminates? But in that case it has to be guaranteed that the client is indeed initialized. All the other methods that are checking the capabilities are also unwrapping.

Reply:
yeah we already unwrap in all other functions I think it's ok to unwrap if you had a doc comment about that above.

Reply:
Done in https://github.com/helix-editor/helix/pull/2507/commits/873c0eef910fc69af43e6431ff40019af03c8b0d